packages/esm-patient-orders-app/src/api/api.ts

```diff
@@ -0,0 +1,136 @@
+ */
+export function useMutatePatientOrders(patientUuid: string) {
+  const { mutate } = useSWRConfig();
+  const mutateOrders = useCallback(
+    () =>
+      mutate((key) => {
+        return typeof key === 'string' && key.startsWith(`/ws/rest/v1/order?patient=${patientUuid}`);
+      }),
+    [patientUuid, mutate],
+  );
+
+  return {
+    mutate: mutateOrders,
+  };
+}
+
+export function getPatientEncounterId(patientUuid: string, abortController: AbortController) {
+  return openmrsFetch(`/ws/rest/v1/encounter?patient=${patientUuid}&order=desc&limit=1&v=custom:(uuid)`, {
+    signal: abortController.signal,
+  });
+}
+
+export function getMedicationByUuid(abortController: AbortController, orderUuid: string) {
+  return openmrsFetch(
+    `/ws/rest/v1/order/${orderUuid}?v=custom:(uuid,route:(uuid,display),action,urgency,display,drug:(display,strength),frequency:(display),dose,doseUnits:(display),orderer,dateStopped,dateActivated,previousOrder,numRefills,duration,durationUnits:(display),dosingInstructions)`,
+    {
+      signal: abortController.signal,
+    },
+  );
+}
+
+export function createEmptyEncounter(
+  patientUuid: string,
+  drugOrderEncounterType: string,
+  activeVisit: Visit | null,
+  sessionLocationUuid: string,
+  abortController?: AbortController,
+) {
+  const now = new Date();
+  const visitStartDate = parseDate(activeVisit?.startDatetime);
+  const visitEndDate = parseDate(activeVisit?.stopDatetime);
+  let encounterDate: Date;
+  if (!activeVisit || (visitStartDate < now && (!visitEndDate || visitEndDate > now))) {
+    now;
+  } else {
+    console.warn(
+      'createEmptyEncounter received an active visit that is not currently active. This is a programming error. Attempting to place the order using the visit start date.',
+    );
+    visitStartDate;
+  }
+  const emptyEncounter = {
+    patient: patientUuid,
+    location: sessionLocationUuid,
+    encounterType: drugOrderEncounterType,
+    encounterDatetime: encounterDate,
+    visit: activeVisit?.uuid,
+    obs: [],
+  };
+
+  return openmrsFetch<OpenmrsResource>('/ws/rest/v1/encounter', {
+    headers: {
+      'Content-Type': 'application/json',
+    },
+    method: 'POST',
+    body: emptyEncounter,
+    signal: abortController?.signal,
+  }).then((res) => res?.data?.uuid);
+}
+
+export function postOrder(body: OrderPost, abortController?: AbortController) {
+  return openmrsFetch(`/ws/rest/v1/order`, {
+    method: 'POST',
+    signal: abortController?.signal,
+    headers: { 'Content-Type': 'application/json' },
+    body,
+  });
+}
+
+export function useOrderEncounter(patientUuid: string): {
+  activeVisitRequired: boolean;
+  isLoading: boolean;
+  error: Error;
+  encounterUuid: string;
+  mutate: Function;
+} {
+  const { systemVisitEnabled, isLoadingSystemVisitSetting, errorFetchingSystemVisitSetting } = useSystemVisitSetting();
+
+  const [nowDateString] = new Date().toISOString().split('T');
```

Comment:
I don't think this is correct since `toISOString()` is [always in UTC](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/toISOString).

This, however, ([stealing from SO](https://stackoverflow.com/a/69963425/274466)) would work:

```js
const [nowDateString] = new Date(new Date().getTime() - new Date().getTimezoneOffset() * 60 * 1000).toISOString().split('T');
```

Or just:

```js
const now = new Date();
// blame Java for the +1
const nowDateString = `${now.getYear()}-${now.getMonth()+1}-${now.getDate()}`
```